packages/testing/examples/useUniswapPrice.test.tsx

```diff
@@ -0,0 +1,76 @@
+import { FixedNumber } from '@ethersproject/bignumber'
+import chai, { expect } from 'chai'
+import { useUniswapPrice, INIT_CODE_HASH, UniswapV2Pair } from '@usedapp/uniswap'
+import { compareAddress } from '@usedapp/core'
+import { renderWeb3Hook } from '../src'
+import { deployMockToken, MOCK_TOKEN_INITIAL_BALANCE } from '../src/utils/deployMockToken'
+import { deployUniswapV2Pair } from '../src/utils/deployMockUniswapV2Pair'
+import { BigNumber } from 'ethers'
+
+chai.use(solidity)
+
+describe('useUniswapPrice', () => {
+  const mockProvider = new MockProvider()
+  const [deployer] = mockProvider.getWallets()
+  const DIGITS = 8
+  const ONE = BigNumber.from(1)
+  const RATIO = BigNumber.from(5)
+  const EXP_SCALE = BigNumber.from(10).pow(DIGITS)
+  let tokenA: Contract
+  let tokenB: Contract
+  let factory: Contract
+  let pair: Contract
+
+  async function addLiquidity(tokenAAmount: BigNumber, tokenBAmount: BigNumber) {
+    await tokenA.transfer(pair.address, tokenAAmount)
+    await tokenB.transfer(pair.address, tokenBAmount)
+    await pair.mint(deployer.address)
+  }
+
+  function sortAddress(tokenA: string, tokenB: string) {
+    return compareAddress(tokenA, tokenB) === -1 ? [tokenA, tokenB] : [tokenB, tokenA]
+  }
+
+  beforeEach(async () => {
+    tokenA = await deployMockToken(deployer)
+    tokenB = await deployMockToken(deployer)
+    ;({ factory, pair } = await deployUniswapV2Pair(deployer, tokenA, tokenB))
+    // RATIO = tokenAReserve / tokenBReserve = 5
+    await addLiquidity(MOCK_TOKEN_INITIAL_BALANCE, MOCK_TOKEN_INITIAL_BALANCE.div(RATIO))
+  })
+
+  it('get init code hash', async () => {
+    const initCodeHash = keccak256(['bytes'], [pack(['bytes'], [`0x${UniswapV2Pair.bytecode}`])])
+    expect(initCodeHash).to.equal(INIT_CODE_HASH)
+  })
+
+  it('compute pair address by using CREATE2', async () => {
+    const [token0Addr, token1Addr] = sortAddress(tokenA.address, tokenB.address)
+    const salt = keccak256(['bytes'], [pack(['address', 'address'], [token0Addr, token1Addr])])
+    const computedAddress = getCreate2Address(factory.address, salt, INIT_CODE_HASH)
+    expect(computedAddress).to.equal(pair.address)
+  })
+
+  it('get price', async () => {
+    const [token0Addr] = sortAddress(tokenA.address, tokenB.address)
+
+    // base/quate (e.g. ETH/DAI): price of baseToken in quateToken = quateTokenReserve / baseTokenReserve
+    const [numerator, denominator] = tokenA.address === token0Addr ? [ONE, RATIO] : [RATIO, ONE]
+    const price = numerator.mul(EXP_SCALE).div(denominator)
+
+    const { result, waitForCurrent } = await renderWeb3Hook(
+      () => useUniswapPrice(tokenA.address, tokenB.address, { factory: factory.address }),
+      {
+        mockProvider,
+      }
+    )
+    await waitForCurrent((val) => val !== undefined)
+    expect(result.error).to.be.undefined
+    expect(result.current?.toString()).to.eq(FixedNumber.fromValue(price, DIGITS).toString())
```

Comment:
ParseFixed() could work here too

Reply:
With or without using BigNumber, all seems well except for the failing test...

BigNumber seems to be preferable as a return value than FixedNumber since many hooks exported usually have it exposed instead and they tend to get returned directly from ethers.js contract calls numeric values

Reply:
I understand what you are saying. So, should the hook return 10^x times the price because otherwise I don't think it will work if the price is fractional?
In other words, if the actual price is 0.2021, should the hook return a BigNumber of 20210000 (where x is 8)?

Reply:
I couldn't fork on my machine to test too, but here's my thought about `BigNumber` vs `FixedNumber`

Behind the scene : 

`FixedNumber.fromValue(price, DIGITS)` will convert the parameters into string with `formatFixed(value, decimals)`,
then the return value get parsed back into a `BigNumber `with `parseFixed(value, decimals)`.

Your implementation works too, this is just a detail anyways.